libmozevent/bus.py

```diff
@@ -57,9 +59,21 @@ async def send(self, name: str, payload: Any):
         if isinstance(queue, RedisQueue):
             redis = await AsyncRedis.connect()
             assert redis is not None
-            await redis.rpush(
-                queue.name, pickle.dumps(payload, protocol=pickle.HIGHEST_PROTOCOL)
-            )
+
+            async def send_with_retry(retry=0):
```

Comment:
Since you're using the built-in support for retry on error, no need to implement it manually here